File: node/src/pickup-points.ts
import { HttpClient } from './http-client';
import type { ShipzyConfig } from './http-client';
import type { ApiResult } from './index';

// ============ Types ============

export type PickupPointType = 'store' | 'locker' | 'counter';
export type PickupPointStatus = 'active' | 'inactive';

export interface PickupPoint {
    id: string;
    merchant_id: string;
    type: PickupPointType;
    name: string;
    address: string;
    contact_phone?: string;
    contact_email?: string;
    opening_hours?: string;
    status: PickupPointStatus;
    latitude?: number;
    longitude?: number;
    country_code?: string;
    created_at: string;
    updated_at: string;
}

export interface CreatePickupPointRequest {
    name: string;
    type?: PickupPointType;
    address?: string;
    contact_phone?: string;
    contact_email?: string;
    opening_hours?: string;
    latitude?: number;
    longitude?: number;
    country_code?: string;
}

export interface PickupPointListResponse {
    data: PickupPoint[];
    total: number;
}

// ============ Client ============

export class PickupPointClient extends HttpClient {
    async list(activeOnly = true): Promise<ApiResult<PickupPointListResponse>> {
        const q = activeOnly ? '' : '?active_only=false';
        return this.request(`/api/v1/admin/pickup-points/${q}`);
    }

    async get(id: string): Promise<ApiResult<PickupPoint>> {
        return this.request(`/api/v1/admin/pickup-points/${id}`);
    }

    async create(req: CreatePickupPointRequest): Promise<ApiResult<PickupPoint>> {
        return this.request('/api/v1/admin/pickup-points/', 'POST', req);
    }

    async update(id: string, updates: Partial<PickupPoint>): Promise<ApiResult<{ status: string }>> {
        return this.request(`/api/v1/admin/pickup-points/${id}`, 'PUT', updates);
    }

    async deactivate(id: string): Promise<ApiResult<{ status: string }>> {
        return this.request(`/api/v1/admin/pickup-points/${id}/deactivate`, 'POST', {});
    }
}

File: python/zymeup_sdk/pickup_points.py
"""Zymeup Python SDK - Pickup Points Client"""
from typing import Optional, Dict, Any, List


class PickupPointClient:
    """Pickup points management client"""

    def __init__(self, client):
        self._client = client

    def list(self, active_only: bool = True) -> Dict[str, Any]:
        """List pickup points"""
        params: Dict[str, Any] = {}
        if not active_only:
            params["active_only"] = "false"
        return self._client.request("GET", "/api/v1/admin/pickup-points/", params=params)

    def get(self, pickup_point_id: str) -> Dict[str, Any]:
        """Get pickup point detail"""
        return self._client.request("GET", f"/api/v1/admin/pickup-points/{pickup_point_id}")

    def create(self, data: Dict[str, Any]) -> Dict[str, Any]:
        """Create a new pickup point"""
        return self._client.request("POST", "/api/v1/admin/pickup-points/", json=data)

    def update(self, pickup_point_id: str, data: Dict[str, Any]) -> Dict[str, Any]:
        """Update a pickup point"""
        return self._client.request("PUT", f"/api/v1/admin/pickup-points/{pickup_point_id}", json=data)

    def deactivate(self, pickup_point_id: str) -> Dict[str, Any]:
        """Deactivate a pickup point"""
        return self._client.request("POST", f"/api/v1/admin/pickup-points/{pickup_point_id}/deactivate")
